### ISSUE TITLE
[Bug] Tab list scroll position resets when using a private tab

### ISSUE THREAD

**Issue by marcel263** [mozilla-mobile/fenix#19943]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

## Steps to reproduce
- Open sufficient normal, non-private tabs so a scrollable tab list/grid shows 
- Scroll to a certain position in normal tab list (the first button from the left which you can do once tab button is pressed)
- Open a private tab
- Reopen the normal tab list by pressing the tabs button
### Expected behavior
Just scrolled to position in tab list is maintained 
### Actual behavior
Resets to top

Please note that this was not tested against the Sync feature which is another possibly affected case.

### Device information

* Samsung android 11 OneUI 3.1
* Nightly 91.0a1 (Build #2015815371)

**Comment by SoftVision-LorandJanos** [comment]
Hi @marcel263 ! Thank you for your report! I can reproduce the issue you described. When you just switch between the private and normal tabs list, the list position remains as left, but as soon as you open a private tab, it resets to the top. I'm not sure if it's intentional this way or not.
Tested on the latest Nightly (91.0a1 2021-06-30T05:09:53.235).
Devices used: Motorola Nexus 6 (Android 7.1.1), Samsung Galaxy Tab 3 (Android 9), Google Pixel 4 (Android 11) and Huawei P9 Lite (Android 7.0)

**Comment by jonalmeida** [comment]
How do you open a private tab without leaving the tabs tray? We don't store the scroll position once you leave the tabs tray, instead, when you first open it, it scrolls to the selected tab (private or normal).

**Comment by marcel263** [comment]
> How do you open a private tab without leaving the tabs tray? We don't store the scroll position once you leave the tabs tray, instead, when you first open it, it scrolls to the selected tab (private or normal).

Having interacted with private tabs, after which returning to normal tab tray, the last known scroll position i.e. selected tab is lost. Always scrolls to the top. 

This behavior of forgetting the normal tab inconveniences common workflow for private tabs that were created from normal tabs ("open link in private tab" long-press menu option), e.g. a quick link check for untrusted domains, reading paywalled sites. Alas, a user might spend more time scrolling back to their last normal tab than they spent interacting with private tab, only becoming more impractical as one's tab count grows.

In short, normal tab position is always reset to the top after opening/creating private tabs while returning to normal tab tray. This behavior is inconsistent from the behavior when user merely switches between modes without any interaction, in that case normal tab tray position is remembered. Remembering last tab position is beneficial for quick switching between modes.

**Comment by marcel263** [comment]
Closed since it is duplicate of #19231
